packages/consumption/src/modules/attributes/local/LocalAttribute.ts

```diff
@@ -172,6 +175,35 @@ export class LocalAttribute extends CoreSynchronizable implements ILocalAttribut
         return typeof this.shareInfo !== "undefined";
     }
 
+    public setDeletionInfo(deletionInfo: LocalAttributeDeletionInfo, ownAddress: CoreAddress): this {
+        if (this.isRepositoryAttribute(ownAddress)) {
+            throw CoreErrors.attributes.cannotSetDeletionInfoOfRepositoryAttributes();
+        }
+
+        if (this.isOwnSharedAttribute(ownAddress) && this.hasPeerSharedAttributeDeletionInfo()) {
+            throw CoreErrors.attributes.invalidDeletionInfoOfOwnSharedAttribute();
+        }
+
+        if (this.isPeerSharedAttribute() && this.hasOwnSharedAttributeDeletionInfo()) {
+            throw CoreErrors.attributes.invalidDeletionInfoOfPeerSharedAttribute();
+        }
+
+        this.deletionInfo = deletionInfo;
+        return this;
+    }
+
+    public hasPeerSharedAttributeDeletionInfo(): boolean {
+        return this.deletionInfo?.deletionStatus === DeletionStatus.DeletedByOwner || this.deletionInfo?.deletionStatus === DeletionStatus.ToBeDeleted;
+    }
+
+    public hasOwnSharedAttributeDeletionInfo(): boolean {
+        return this.deletionInfo?.deletionStatus === DeletionStatus.DeletedByPeer || this.deletionInfo?.deletionStatus === DeletionStatus.ToBeDeletedByPeer;
+    }
+
+    public hasDeletionInfo(): boolean {
```

Comment:
just a small reminder - no direct CTA: this could be a typeguard, which could help you later on (or maybe also now?) when accessing the deletion info.